### DISCUSSION TITLE
Why comparison is made between roleOrgMap's Id with rootOrgId

### DISCUSSION THREAD

**Discussion by Xitija** [Sunbird-Ed/Community#585] [Sunbird ED configuration discussion] [answered]
Hello All,

I am trying to enable the `invite-only` feature on Sunbird-4.10.
Till now I have completed calling API and making changes as mentioned in this ticket [Link here](https://github.com/orgs/Sunbird-Ed/discussions/557#discussioncomment-6350834)

I am unable to understand the following code written in course-batch.service.ts in the method getUserList
```
      const mentorOrg = this.userService.userProfile.roleOrgMap['CONTENT_CREATOR'];
      if (mentorOrg && mentorOrg.includes(this.userService.rootOrgId)) {
        option.data.request.filters['rootOrgId'] = this.userService.rootOrgId;
      } else if (mentorOrg) {
        option.data.request.filters['organisations.organisationId'] = mentorOrg;
      }
      // option.data.request.filters['organisations.roles'] = ['COURSE_MENTOR'];  < I have commented this line because I want all the users and not just mentor list>

```
I want to understand why is the if condition is checking mentorOrg with rootOrgId of user
What exactly is roleOrgMap['CONTENT_CREATOR'], Why is it used? What does it signify?
Why is the mentorOrg given more preference than the rootOrg?

courseBatchservice.ts  (contains api call method - getUserList - Line 37) -
[Source Code File](https://github.com/Sunbird-Ed/SunbirdEd-portal/blob/release-4.10.2_RC12/src/app/client/src/app/modules/learn/services/course-batch/course-batch.service.ts)

@coolbung @rajeevsathish @vinod303

**Comment by rajeevsathish** [comment, marked as answer]
@Xitija this is the code to get users from the org the course is created for.

please put the breakpoints in the browser and follow the code you will be able to understand what is the code doing.

cc: @coolbung @vinod303

**Reply by Xitija** [reply]
For now we are considering the code to work in the first case only
